presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/EliminateCrossJoins.java

```diff
@@ -135,6 +136,78 @@ public static boolean isOriginalOrder(List<Integer> joinOrder)
                 .collect(toImmutableList());
     }
 
+    public static PlanNode createJoinsReplacement(PlanNode node, JoinGraph graph, List<Integer> joinOrder, PlanNodeIdAllocator idAllocator)
```

Comment:
Maybe call this method "buildJoinTree", which more closely captures what the method does.